src/main/scala/it/pagopa/interop/purposeprocess/api/impl/PurposeApiServiceImpl.scala

```diff
@@ -100,10 +100,11 @@ final case class PurposeApiServiceImpl(
     val result: Future[Purpose] = for {
       organizationId <- getOrganizationIdFutureUUID(contexts)
       _              <- assertOrganizationIsAConsumer(organizationId, seed.consumerId)
-      tenant         <- tenantManagementService.getTenant(organizationId)
-      tenantKind     <- tenant.kind.toFuture(TenantKindNotFound(tenant.id))
-      clientSeed     <- seed.apiToDependency(schemaOnlyValidation = true)(tenantKind).toFuture
-      agreements     <- agreementManagementService.getAgreements(
+      _      <- Future.failed(MissingFreeOfChargeReason) whenA (seed.isFreeOfCharge && seed.freeOfChargeReason.isEmpty)
```

Comment:
don't use infix notation please
also, `whenA` is a cats method, and as such should not be used with Futures

Reply:
Martin Odersky loves infix notation <img src="https://media4.giphy.com/media/4N1wOi78ZGzSB6H7vK/giphy.gif"/>